auth-web/src/components/datatable/BaseVDataTable.vue

```diff
@@ -0,0 +1,277 @@
+    loading: { default: false },
+    loadingText: { default: 'Loading...' },
+    noDataText: { default: 'No results found.' },
+    resetFilters: { default: false },
+    setItems: { default: [] as object[] },
+    setHeaders: { default: [] as BaseTableHeaderI[] },
+    totalItems: { type: Number }
+  },
+  setup (props, { emit }) {
+    // static vars (for initializing)
+    const ITEMS_PER_PAGE = 5
+    const PAGINATION_COUNTER_STEP = 4
+    const INITIAL_ITEMS_PER_PAGE_OPTIONS = [...Array(PAGINATION_COUNTER_STEP)].map((value, index) => ITEMS_PER_PAGE * (index + 1))
+    // reactive vars
+    const state = (reactive({
+      filtering: false,
+      headers: _.cloneDeep(props.setHeaders),
+      sortedItems: [...props.setItems],
+      tableDataOptions: props.initialTableDataOptions
+    }) as unknown) as BaseTableStateI
+
+    const filter = _.debounce(async (header: BaseTableHeaderI) => {
+      // rely on custom filterApiFn to alter result set if given (meant for server side filtering)
+      if (header.customFilter.filterApiFn) {
+        state.filtering = true
+        await header.customFilter.filterApiFn(header.customFilter.value)
+        state.filtering = false
+      }
+    }, 500)
+
+    watch(() => props.setItems, (val: object[]) => { state.sortedItems = [...val] })
+    watch(() => props.setHeaders, (val: BaseTableHeaderI[]) => {
+      // maintain filters
+      const filters = {}
+      state.headers.forEach((header) => {
+        if (header.hasFilter) filters[header.col] = header.customFilter.value
+      })
+      val.forEach((header) => {
+        if (Object.keys(filters).includes(header.col)) header.customFilter.value = filters[header.col]
+      })
+      state.headers = val
+    })
+    watch(() => props.clearFilters, () => {
+      state.headers.forEach((header) => {
+        if (header.hasFilter) header.customFilter.value = ''
+      })
+    })
+    watch(() => state.tableDataOptions, (val: DataOptions) => { emit('update-table-options', val) })
+
+    return {
+      INITIAL_ITEMS_PER_PAGE_OPTIONS,
+      filter,
+      ...toRefs(state)
+    }
+  }
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/scss/theme.scss';
+.base-table {
+
+  &__header {
+
+    &__filter {
+      padding-top: 10px;
+
+      &__select,
+      &__textbox {
+        font-size: 0.825rem !important;
+
+        .v-input__control .v-input__slot .v-input__append-inner .v-input__icon.v-input__icon--clear .v-icon {
+          color: $app-blue;
+        }
+      }
+    }
+
+    &__title {
+      color: $gray9 !important;
+      font-size: 0.875rem;
+    }
+
+  }
+
+  &__item {
+    border: 0px;
+    font-size: 0.875rem;
+    padding-top: 0.5rem;
+    padding-bottom: 0.5rem;
+    vertical-align: top;
+  }
+
+  &__text {
+    border: 0px;
+    position: sticky;
+    left: 0;
+    flex-grow: 0;
+    flex-shrink: 0;
+  }
+
+  ::v-deep .v-data-footer {
+    min-width: 100%;
+  }
+
+  ::v-deep .v-data-table__wrapper::-webkit-scrollbar {
```

Comment:
going to hurt a bit upgrading vuetify with these :(

Reply:
yeah I tried to keep as much of the existing styling as possible. Can't really get around it while depending on vuetify components so heavily